crates/ruff_python_semantic/src/cfg/graph.rs

```diff
@@ -0,0 +1,282 @@
+use ruff_index::{newtype_index, IndexVec};
+use ruff_python_ast::Stmt;
+use ruff_text_size::{Ranged, TextRange};
+use smallvec::{smallvec, SmallVec};
+
+/// Returns the control flow graph associated to an array of statements
+pub fn build_cfg(stmts: &[Stmt]) -> CFG<'_> {
+    let mut builder = CFGBuilder::with_capacity(stmts.len());
+    builder.process_stmts(stmts);
+    builder.finish()
+}
+
+/// Control flow graph
+#[derive(Debug)]
+pub struct CFG<'stmt> {
```

Comment:
Nit: The all uppercase breaks my mind but I'm not sure why. I'm sort of leaning towards avoiding the `cfg` abbreviation and just using the full name `ControlFlowGraph` (also less confusing with rusts `cfg!` macro

Reply:
Ha, I think the module name is fine... I know, I don't make much sense